src/lib/plugin_apis/fs.api

```diff
@@ -147,4 +196,77 @@ BDFSExt4Info* bd_fs_ext4_get_info (gchar *device, GError **error);
  */
 gboolean bd_fs_ext4_resize (gchar *device, guint64 new_size, GError **error);
 
+/**
+ * bd_fs_xfs_mkfs:
+ * @device: the device to create a new xfs fs on
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether a new xfs fs was successfully created on @device or not
+ */
+gboolean bd_fs_xfs_mkfs (gchar *device, GError **error);
+
+/**
+ * bd_fs_xfs_wipe:
+ * @device: the device to wipe an xfs signature from
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether an xfs signature was successfully wiped from the @device or
+ *          not
+ */
+gboolean bd_fs_xfs_wipe (gchar *device, GError **error);
+
+/**
+ * bd_fs_xfs_check:
+ * @device: the device the file system on which to check
```

Comment:
Something like "the device containing the file system to check" would be clearer.